src/spatial/utils.jl

```diff
@@ -0,0 +1,217 @@
+using LightGraphs
+"""
+A file with types, structs and functions for spatial simulations
+"""
+
+"""
+stores info for a spatial jump
+"""
+struct SpatialJump{J}
+    src::J    # source location
+    jidx::Int # index of jump as a hop or reaction
+    dst::J    # destination location
+end
+
+############ abstract spatial system struct ##################
+"""
+Contains all info about the topology of the system
+"""
+abstract type AbstractSpatialSystem end
+
+# Interface:
+# num_sites(spatial_system) = total number of sites
+# neighbors(spatial_system, site) = an iterator over the neighbors of site
+# num_neighbors(spatial_system, site) = number of neighbors of site
+
+################### LightGraph ########################
+num_sites(graph::AbstractGraph) = LightGraphs.nv(graph)
```

Comment:
If you are getting name clashes so you need to qualify with `LightGraphs`, you might as well just `import LightGraphs` instead of `using`.